### ISSUE TITLE
[V2] Thrown app:preDebugBuild against app:debugRuntimeClasspath

### ISSUE THREAD

**Issue by santanu4ver** [wix/react-native-navigation#3641]
### Issue Description

`react-native run-android` throws following error and terminates:

```
FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':app:preDebugBuild'.
> Could not resolve all task dependencies for configuration ':app:debugRuntimeClasspath'.
   > Could not resolve project :react-native-navigation.
     Required by:
         project :app
      > Unable to find a matching configuration of project :react-native-navigation: None of the consumable configurations have attributes.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1s
```

### Steps to Reproduce / Code Snippets / Screenshots

This one is a app:preDebugBuild error as described at [here](https://wix.github.io/react-native-navigation/v2/#/docs/Installing?id=android). But taking documented measure didn't fix the problem.

- Create a new RN project
- Follow all the steps as described [here](https://wix.github.io/react-native-navigation/v2/#/docs/Installing?id=android) for Android run
- Run `react-native run-android` on Terminal

---
### Environment

Versions uses in the RN application:

```
"dependencies": {
    "react": "16.4.1",
    "react-native": "0.56.0",
    "react-native-navigation": "^2.0.2431"
  }
```

OS Platform: macOS El Capitan
Device Simulator: Android
Gradle: 4.6
Java: 1.8.0_181
Node: 10.7.0

### Example Project

Download from [Dropbox](https://www.dropbox.com/s/8rrdzvnjtogehla/rnnv2.zip?dl=0)

**Comment by ujwal-setlur** [comment]
@santanu4ver, I downloaded the example project you posted on your dropbox. I see several errors in build.gradle files and the MainApplication.java and MainActivity.java files. Are you sure you followed the instructions correctly? I made changes to get the app to compile but it doesn't launch because some image files are missing and a few other things.

I suspect what you posted is a mixture of RNN v1 and v2. Frankly, the gradle and java files were just wrong. I suggest you verify the steps you followed. I am happy to post the package with the changes I made.

**Comment by ujwal-setlur** [comment]
Here are my changes:

```diff
diff -ru rnnv2/android/app/build.gradle rnnv2-changes/android/app/build.gradle
--- rnnv2/android/app/build.gradle	2018-07-27 21:19:11.000000000 -0700
+++ rnnv2-changes/android/app/build.gradle	2018-07-28 13:00:20.000000000 -0700
@@ -94,21 +94,25 @@
 def enableProguardInReleaseBuilds = false

 android {
-    compileSdkVersion 25
-    buildToolsVersion "27.0.3"
+    compileSdkVersion rootProject.ext.compileSdkVersion
+    buildToolsVersion rootProject.ext.buildToolsVersion

     defaultConfig {
         applicationId "com.rnnv2"
-        minSdkVersion 19
-        targetSdkVersion 25
+        minSdkVersion rootProject.ext.minSdkVersion
+        targetSdkVersion rootProject.ext.targetSdkVersion
+        missingDimensionStrategy "RNN.reactNativeVersion", "reactNative56"
         versionCode 1
         versionName "1.0"
         ndk {
             abiFilters "armeabi-v7a", "x86"
         }

-        missingDimensionStrategy "RNN.reactNativeVersion", "reactNative51"
     }
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+     }
     splits {
         abi {
             reset()
diff -ru rnnv2/android/app/src/main/java/com/rnnv2/MainActivity.java rnnv2-changes/android/app/src/main/java/com/rnnv2/MainActivity.java
--- rnnv2/android/app/src/main/java/com/rnnv2/MainActivity.java	2018-07-27 09:04:27.000000000 -0700
+++ rnnv2-changes/android/app/src/main/java/com/rnnv2/MainActivity.java	2018-07-28 12:56:21.000000000 -0700
@@ -4,12 +4,4 @@

 public class MainActivity extends NavigationActivity {

-    /**
-     * Returns the name of the main component registered from JavaScript.
-     * This is used to schedule rendering of the component.
-     */
-    @Override
-    protected String getMainComponentName() {
-        return "rnnv2";
-    }
 }
diff -ru rnnv2/android/app/src/main/java/com/rnnv2/MainApplication.java rnnv2-changes/android/app/src/main/java/com/rnnv2/MainApplication.java
--- rnnv2/android/app/src/main/java/com/rnnv2/MainApplication.java	2018-07-27 09:06:03.000000000 -0700
+++ rnnv2-changes/android/app/src/main/java/com/rnnv2/MainApplication.java	2018-07-28 13:06:44.000000000 -0700
@@ -1,57 +1,45 @@
 package com.rnnv2;

-import android.app.Application;
+import android.support.annotation.Nullable;

-import com.facebook.react.ReactApplication;
-import com.facebook.react.ReactNativeHost;
 import com.facebook.react.ReactPackage;
-import com.facebook.react.shell.MainReactPackage;
-import com.facebook.soloader.SoLoader;
+import com.facebook.react.ReactNativeHost;
+
 import com.reactnativenavigation.NavigationApplication;
+import com.reactnativenavigation.react.NavigationReactNativeHost;
+// the following line is present so that `react-native link` works correctly
+// import com.facebook.react.ReactApplication;

 import java.util.Arrays;
 import java.util.List;

 public class MainApplication extends NavigationApplication {
-
-  private final ReactNativeHost mReactNativeHost = new ReactNativeHost(this) {
     @Override
-    public boolean getUseDeveloperSupport() {
-      return BuildConfig.DEBUG;
+    public boolean isDebug() {
+        return BuildConfig.DEBUG;
     }

-    @Override
     protected List<ReactPackage> getPackages() {
-      return Arrays.<ReactPackage>asList(
-          new MainReactPackage()
-      );
+         // Add additional packages you require here
+         return Arrays.<ReactPackage>asList(
+            // the following line is present so that `react-native link` works correctly
+            // new MainReactPackage(),
+         );
     }

+
     @Override
-    protected String getJSMainModuleName() {
-      return "index";
+    public List<ReactPackage> createAdditionalReactPackages() {
+        return getPackages();
     }
-  };
-
-  @Override
-  public ReactNativeHost getReactNativeHost() {
-    return mReactNativeHost;
-  }

   @Override
-  public void onCreate() {
-    super.onCreate();
-    SoLoader.init(this, /* native exopackage */ false);
+  protected ReactNativeHost createReactNativeHost() {
+    return new NavigationReactNativeHost(this) {
+      @Override
+      protected String getJSMainModuleName() {
+        return "index";
+      }
+    };
   }
-
-  @Override
-    public boolean isDebug() {
-        return BuildConfig.DEBUG;
-    }
-
-    @Nullable
-    @Override
-    public List<ReactPackage> createAdditionalReactPackages() {
-        return null;
-    }
 }
diff -ru rnnv2/android/build.gradle rnnv2-changes/android/build.gradle
--- rnnv2/android/build.gradle	2018-07-27 08:57:55.000000000 -0700
+++ rnnv2-changes/android/build.gradle	2018-07-28 12:59:04.000000000 -0700
@@ -31,7 +31,7 @@

 ext {
     buildToolsVersion = "26.0.3"
-    minSdkVersion = 16
+    minSdkVersion = 19
     compileSdkVersion = 26
     targetSdkVersion = 26
     supportLibVersion = "26.1.0"
diff -ru rnnv2/android/settings.gradle rnnv2-changes/android/settings.gradle
--- rnnv2/android/settings.gradle	2018-07-27 21:20:22.000000000 -0700
+++ rnnv2-changes/android/settings.gradle	2018-07-28 12:49:40.000000000 -0700
@@ -3,4 +3,4 @@
 include ':app'

 include ':react-native-navigation'
-project(':react-native-navigation').projectDir = new File(rootProject.projectDir, '../../lib/android/app/')
\ No newline at end of file
+project(':react-native-navigation').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-navigation/lib/android/app/')
```

**Comment by santanu4ver** [comment]
This is possible that I might made some mistakes (I'm mostly newbie to RN) but I'm wondering about the mix of v1 and v2 mixed-up. I believe I modified by creating a fresh project, but it's messed anyway.

After following your changes to a fresh created project, this time I able to complete the compilation. I copied 'playground' source from v2 repository to the project, and I able to get them run, too! 

Thank you! You really made my day ) It's irritating when something breaks. 

I'm closing this issue.
